src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexer.cs

```diff
@@ -458,10 +458,16 @@ internal static string GetSharedListenerIdOrNull(ITriggerBinding triggerBinding)
 
         internal static void CheckRetrySupport(ITriggerBinding triggerBinding, FunctionDescriptor descr, ILogger logger)
         {
-            if (descr.RetryStrategy != null && triggerBinding != null && triggerBinding.GetType().GetCustomAttribute<SupportsRetryAttribute>() == null)
+            if (descr.RetryStrategy != null)
             {
-                logger?.LogWarning($"Retries are not supported by the trigger binding for { descr.ShortName}.");
-                descr.RetryStrategy = null; // Need to fix tests
+                logger?.LogDebug($"Retry strategy is defined for function '{descr?.ShortName}' with trigger binding type '{triggerBinding?.GetType().Name}'. Strategy = '{JsonConvert.SerializeObject(descr?.RetryStrategy)}'");
```

Comment:
nit: Use log params instead of string interpolation - https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loggerextensions.loginformation?view=dotnet-plat-ext-3.1